app/screens/box/box-add-update-screen.tsx

```diff
@@ -89,37 +118,64 @@ export const BoxAddUpdateScreen: React.FC<Props> = ({ navigation, route }) => {
       navigation.pop()
     } catch (error) {
       console.log(error)
-      Alert.alert('Error', 'Make sure the address format is correct!')
     } finally {
       pressed.current = false
     }
   }
-  const checkConnection = async () => {
+
+  const newFulaClient = async () => {
     if (dIDCredentials?.username && dIDCredentials?.password && form.address) {
-      setTestinConnection(true)
       const keyPair = helper.getMyDIDKeyPair(dIDCredentials.username, dIDCredentials.password)
       try {
-        const newClient = await fula.newClient(
+        await fula.shutdown()
```

Comment:
if fula is not initialized this might return exception adn not go to next line. that is the reason I think we should check with isReady before shutting down
if await isReady() then shutdown() else no need

Reply:
You are right, I will check the isReady before newClient,
Also, we can handle the exception in the shutdown method when fula is not ready inside the native code.